### DISCUSSION TITLE
[Question] Get information of change of a model

### DISCUSSION THREAD

**Discussion by chengtie** [microsoft/monaco-editor#3657] [Q&A] [answered]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

I use Monaco Editor and [React Monaco Editor](https://github.com/react-monaco-editor/react-monaco-editor) to build our own editor.

An event `onChange`  is emitted when the content of the current model has changed, where we can get `newValue` of the model.

Now, I would like to get information of comparison of newValue and oldValue: for instance, insertion of certain characters at a location, deletion of certain characters at a location, etc.

Does anyone know how to get these information?

**Comment by zhangbaojia** [comment]
I also want to know.

**Comment by ted-marozzi** [comment, marked as answer]
Perhaps `editor.onDidChangeModelContent((a) => console.log(a));` is what you want

```json
{
    "changes": [
        {
            "range": {
                "startLineNumber": 9,
                "startColumn": 6,
                "endLineNumber": 9,
                "endColumn": 6
            },
            "rangeLength": 0,
            "text": "o",
            "rangeOffset": 74,
            "forceMoveMarkers": false
        }
    ],
    "eol": "\n",
    "versionId": 7,
    "isUndoing": false,
    "isRedoing": false,
    "isFlush": false
}
```
